src/platform/web/ui/general/LazyListView.js

```diff
@@ -0,0 +1,128 @@
+import {mountView} from "./utils.js";
+import {insertAt, ListView} from "./ListView.js";
+
+class Range {
+    constructor(start = 0, end = 0) {
+        this.start = start;
+        this.end = end;
+        this._expanded = false;
+    }
+
+    _onInitialExpand() {
+        if (this._expanded) { return; }
+        this._initialStart = this.start;
+        this._expanded = true;
+    }
+
+    expandFromEnd(units) {
+        this._onInitialExpand();
+        this.start = this.end;
+        this.end += units;
+        this._expanded = true;
+    }
+
+    contains(idx) {
+        const start = this._expanded ? this._initialStart : this.start;
+        return idx >= start && idx <= this.end;
+    }
+}
+
+export class LazyListView extends ListView {
+    constructor({itemHeight, height, appendCount = 5, ...options}, childCreator) {
+        super(options, childCreator);
+        this._itemHeight = itemHeight;
+        this._height = height;
+        this._appendCount = appendCount;
+        this._range = new Range();
+    }
+
+    _isFullyScrolled() {
+        return this._root.scrollHeight - Math.abs(this._root.scrollTop) === this._root.clientHeight;
+    }
+
+    _renderMoreIfNeeded() {
+        if (!this._isFullyScrolled()) {
+            return;
+        }
+        this._range.expandFromEnd(this._appendCount);
```

Comment:
I think this will add `n` (default of 5) items *every* time you scroll a pixel? I'd like to interpret this number as a padding rather, where you have the range the corresponds to the current viewport and you add `n` items on either end (limited by 0 and the length of the list on either end of course).